ibis/sql/compiler.py

```diff
@@ -1010,10 +1030,28 @@ def flatten_union(table):
     return [table]
 
 
+def flatten_intersection(table):
```

Comment:
can you type things

Reply:
@jreback There's no type hinting anywhere else in this repo, so wouldn't it be weird to have it here?

Reply:
@jreback I would love to see type hint support on this package though, so if that is what you mean I'm happy to add them